AirlineApp.Infrastructure.RabbitMq/AirlineAppRabbitMqConsumer.cs

```diff
@@ -56,20 +82,55 @@ protected override Task ExecuteAsync(CancellationToken stoppingToken)
     /// <returns>A task that represents the asynchronous message processing operation.</returns>
     private async Task ReceiveMessage(BasicDeliverEventArgs args, CancellationToken stoppingToken)
     {
-        _logger.LogInformation("Received a message from queue {queue}", _queueName);
         try
         {
+            _logger.LogDebug("Received a message from queue {Queue}", _queueName);
+
             stoppingToken.ThrowIfCancellationRequested();
-            var contracts = JsonSerializer.Deserialize<List<TicketEditDto>>(new MemoryStream(args.Body.ToArray()))
+
+            var contracts = JsonSerializer.Deserialize<List<TicketEditDto>>(args.Body.Span)
                 ?? throw new FormatException("Unable to parse contracts from message body");
 
             using var scope = _scopeFactory.CreateScope();
             var ticketService = scope.ServiceProvider.GetRequiredService<ITicketService>();
+
             await ticketService.ReceiveContractList(contracts);
+
+            _logger.LogDebug("Successfully processed batch of {Count} contracts", contracts.Count);
+        }
+        catch (OperationCanceledException)
+        {
+            _logger.LogInformation("Message processing was cancelled");
+            throw;
         }
         catch (Exception ex)
         {
-            _logger.LogError(ex, "Exception occurred during receiving contracts from {queue}", _queueName);
+            _logger.LogError(ex, "Exception occurred during processing contracts from {Queue}", _queueName);
+        }
+    }
+
+    /// <summary>
+    /// Disposes the resources used by the consumer.
+    /// </summary>
+    protected virtual void Dispose(bool disposing)
+    {
+        if (!_disposed)
+        {
+            if (disposing)
+            {
+                _channel?.Close();
+                _channel?.Dispose();
+            }
+            _disposed = true;
         }
     }
+
+    /// <summary>
+    /// Disposes the consumer.
+    /// </summary>
+    public override void Dispose()
+    {
+        Dispose(true);
+        GC.SuppressFinalize(this);
+    }
```

Comment:
По сути вы переопределили Dispose и для BackgroundService. Лучше добавить еще вызов базового Dispose.
```suggestion
    public override void Dispose()
    {
        base.Dispose();          // Сначала базовый класс
        Dispose(true);           // Потом свой
        GC.SuppressFinalize(this);
    }
```
Либо же можете оставить так но base.Dispose() вызвать в вашем методе выше.
Также хочу заметить что у вас помимо IModel что реализует IDisposable имеется IConnection которое тоже по хорошему закрыть и освободить.